### DISCUSSION TITLE
Memory management for pyserini.encode

### DISCUSSION THREAD

**Discussion by noamaon** [castorini/pyserini#2026] [Q&A]
Hi, I am trying to use a HF e5 model to encode the msmarco-vs-passage dataset. In the process I split the data into 8 shards, and encode each shard using an Nvidia a100 GPU and 100GB of RAM with a batch size of 32. I am getting oom errors when running the scripts. Is there a different way to preform the encoding? Is there an argument for managing memory?

**Comment by octocat20** [comment]
No special memory flag on pyserini.encode. This is almost always GPU VRAM, not your 100GB system RAM.

Useful knobs:
- drop --batch-size (try 8 or 4)
- add --fp16
- lower --max-length if you can
- keep using --shard-id / --shard-num

Sharding only splits the corpus. One shard can still OOM if the batch is fat.

Run a single shard, watch nvidia-smi, and walk the batch size down until it fits. Big host RAM does not help if the A100 is full.
